### DISCUSSION TITLE
can't seem to reconnect to Deepgram via Websocket without refreshing page

### DISCUSSION THREAD

**Discussion by BarnoldSporrance** [deepgram/community#379] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

We’re trying to use Deepgram to transcribe live speech via the browser’s microphone.

The functionally we are building is:
 -  when a user mouses-down on a  Mic button, the connection to Deepgram via web socket opens.  
 - The user speaks while mouse-down, 
 - After finishing the utterance, mouses-up, releasing the mic button. 

At this point, we want the web socket to disconnect.   After that, **without refreshing the browser,** we want the user to be able to mouse-down again, reconnecting to Deepgram, and passing in utterance 2, and so on.

Problem: the above works fine for  Utterance 1, but when we reconnect again to Deepgram for processing of Utterance 2, the reconnection to Deepgram is successful. However, Deepgram does not seem to process utterance 2. It’s not throwing any errors.

Does anyone have any experience with this?  

Thanks very much in advance.

**Comment by lukeocodes** [comment, marked as answer]
Connecting and disconnecting from the websocket can sometimes leave unresolved state on the socket. You might have to delete the socket instance and create it altogether, to ensure there is no unresolved state and/or avoid a memory leak.

Without seeing the code, it is very difficult to advise

**Reply by BarnoldSporrance** [reply]
Hi @lukeocodes,

Sorry for the late reply - wanted to say thanks for the answer. You put us in the right direction, and we worked it out. Thanks again!
